src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -68,17 +71,42 @@ internal class RenameEndpoint : IRenameEndpoint
             return new RegistrationExtensionResult(ServerCapability, options);
         }
 
-        public async Task<WorkspaceEdit?> Handle(RenameParamsBridge request, CancellationToken cancellationToken)
+        /// <inheritdoc/>
+        protected override string EndpointName => LanguageServerConstants.RazorRenameEndpointName;
+
+        /// <inheritdoc/>
+        protected override Task<WorkspaceEdit?> HandleWithoutSingleServerAsync(RenameParamsBridge request, DocumentContext documentContext, RazorCodeDocument codeDocument, SourceText sourceText, CancellationToken cancellationToken)
+            => Task.FromResult<WorkspaceEdit?>(null);
+
+        /// <inheritdoc/>
+        protected override Task<WorkspaceEdit?> HandleInRazorAsync(RenameParamsBridge request, DocumentContext documentContext, RazorCodeDocument codeDocument, SourceText sourceText, CancellationToken cancellationToken)
+            => Task.FromResult<WorkspaceEdit?>(null);
+
+        /// <inheritdoc/>
+        protected override DelegatedRenameParams CreateDelegatedParams(RenameParamsBridge request, DocumentContext documentContext, Projection projection)
+            => new DelegatedRenameParams(
+                documentContext.Identifier,
+                projection.Position,
+                projection.LanguageKind,
+                request.NewName);
+
+        /// <inheritdoc/>
+        protected override async Task<WorkspaceEdit?> RemapResponseAsync(WorkspaceEdit? response, RazorCodeDocument codeDocument, CancellationToken cancellationToken)
         {
-            if (request is null)
+            if (response is null)
             {
-                throw new ArgumentNullException(nameof(request));
+                throw new ArgumentNullException(nameof(response));
             }
 
-            if (!_languageServerFeatureOptions.SupportsFileManipulation)
+            return await _documentMappingService.RemapWorkspaceEditAsync(response, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<WorkspaceEdit?> Handle(RenameParamsBridge request, CancellationToken cancellationToken)
```

Comment:
I don't see anything here worth having this override Handle, or indeed having Handle even be protected.

I think the file type check is a bug, and I should have removed it, and hence I think this method can just be the HandleRazor method.

Reply:
Removed. See what you think